src/ydata_synthetic/synthesizers/regular/pategan/model.py

```diff
@@ -0,0 +1,141 @@
+from math import log
+from typing import List
+
+from tensorflow import GradientTape, concat, expand_dims, ones_like, zeros, zeros_like
+from tensorflow.dtypes import cast, float64
+from tensorflow.keras.losses import BinaryCrossentropy
+from tensorflow.keras.optimizers import Adam
+from tensorflow.random import uniform
+from tensorflow_probability import distributions
+
+from ydata_synthetic.synthesizers import TrainParameters
+from ydata_synthetic.synthesizers.gan import BaseModel
+
+
+class PATEGAN(BaseModel):
+
+    # pylint:disable=R0913
+    def train(self, data, class_ratios, train_arguments: TrainParameters, num_cols: List[str], cat_cols: List[str]):
+        """
```

Comment:
I'll leave personal comment(s) on the method that would be good to validate.
A general comment is that it seems that this implementation is meant to be conditional where in the article this is not expected.
Neither the generator or discriminators are expected to receive any condition. Ill bring it up in the respective places.